### PR TITLE
Tabs typescript

### LINKED ISSUE #561: Refactor Navigation Tabs to Typescript
Create typings for [Navigation Tabs](https://eds.equinor.com/components/navigation-tabs/)

### PR DESCRIPTION
**Pull request by mimarz**
resolves #561

### CODE REVIEW

File: libraries/core-react/src/Tabs/TabList.tsx [hunk 1]
```diff
@@ -1,38 +1,61 @@
...
 `
 
-const TabList = forwardRef(function TabsList({ children, ...props }, ref) {
-  const { activeTab, handleChange, tabsId, variant, tabsFocused } = useContext(
-    TabsContext,
-  )
+type Props = {
+  /** Sets the width of the tabs */
+  variant?: Variants
+} & React.HTMLAttributes<HTMLDivElement>
```
Comment (pomfrida) [inline libraries/core-react/src/Tabs/TabList.tsx:39]: `children` should be mentioned here as it is required, right? 🤔

File: libraries/core-react/src/Tabs/TabPanel.tsx [hunk 2]
```diff
@@ -25,26 +25,20 @@ const StyledTabPanel = styled.div.attrs(() => ({
   },
 })
 
-const TabPanel = forwardRef(function TabPanel({ ...props }, ref) {
+type Props = {
+  /** If `true`, the panel will be hidden. */
+  hidden?: boolean
+} & React.HTMLAttributes<HTMLDivElement>
```
Comment (pomfrida) [inline libraries/core-react/src/Tabs/TabPanel.tsx:31]: Same here, add `children` as required

File: libraries/core-react/src/Tabs/TabPanels.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React, { forwardRef, ReactElement, useContext } from 'react'
+import { TabsContext } from './Tabs.context'
+
+type Props = React.HTMLAttributes<HTMLDivElement>
```
Comment (pomfrida) [inline libraries/core-react/src/Tabs/TabPanels.tsx:4]: Same here, add `children` as required

**Review by pomfrida — CHANGES_REQUESTED** [review]
LGTM, but I **think** we have to mention the `children` as long as they are required in the component

File: libraries/core-react/src/Tabs/TabList.tsx (hunk 1 above)
Comment (mimarz) [reply libraries/core-react/src/Tabs/TabList.tsx:39]: As discussed during standup. Added tests and fallbacks for if children is not defined to stay as close as possible to native behaviour

**Review by pomfrida — APPROVED** [review]
LGTM 👍

**Review by wenche — APPROVED** [review]
LGTM 👍

**Pull request merged**